### PR TITLE
refactor (ui): delete declare var and refactor some functions

### PR DESCRIPTION
**Pull request by bnjjj**
Signed-off-by: Benjamin Coenen <benjamin.coenen@hotmail.com>

### CODE REVIEW

File: ui/src/app/views/application/show/workflow/tree/item/application.workflow.item.component.ts [hunk 1]
```diff
@@ -319,11 +320,14 @@ export class ApplicationWorkflowItemComponent implements DoCheck {
                 this.oldPipelineStatus = this.workflowItem.pipeline.last_pipeline_build.status;
             }
 
-            if (this.oldPipelineStatus === 'Building' && this.oldPipelineStatus !== this.workflowItem.pipeline.last_pipeline_build.status) {
+            if (this.oldPipelineStatus === 'Building' && this.oldPipelineStatus &&
+                this.oldPipelineStatus !== this.workflowItem.pipeline.last_pipeline_build.status &&
+                    this.oldBranch && this.branch === this.oldBranch) {
```
Comment (fsamin) [inline ui/src/app/views/application/show/workflow/tree/item/application.workflow.item.component.ts]: Could you explain why you need this.branch and why did you change the condition ?

File: ui/src/app/views/application/show/workflow/application.workflow.html [hunk 2]
```diff
@@ -47,7 +47,8 @@
          [class.horizontal]="workflowOrientation === 'horizontal'"
          [class.vertical]="workflowOrientation === 'vertical'"
          [class.hierarchical]="workflowOrientation === 'hierarchical'">
-        <app-application-tree-workflow *ngIf="application.workflows" [project]="project" [application]="application" [workflowItems]="application.workflows" [orientation]="workflowOrientation" [applicationFilter]="applicationFilter"></app-application-tree-workflow>
+        <app-application-tree-workflow *ngIf="application.workflows" [project]="project" [application]="application" [workflowItems]="application.workflows" [orientation]="workflowOrientation" [applicationFilter]="applicationFilter"
+        [branch]="applicationFilter.branch"></app-application-tree-workflow>
```
Comment (sguiheux) [inline ui/src/app/views/application/show/workflow/application.workflow.html]: no need, applicaitonFilter is already sent

File: ui/src/app/views/application/show/workflow/tree/application.tree.workflow.html [hunk 3]
```diff
@@ -1,6 +1,6 @@
 <ul>
     <li *ngFor="let item of workflowItems; let i = index" [class.last]="i === workflowItems.length -1">
-        <app-application-workflow-item [project]="project"  [workflowItem]="item" [orientation]="orientation" [application]="application" [applicationFilter]="applicationFilter"></app-application-workflow-item>
-        <app-application-tree-workflow *ngIf="item.subPipelines" [project]="project"  [workflowItems]="item.subPipelines" [orientation]="orientation" [application]="application" [applicationFilter]="applicationFilter"></app-application-tree-workflow>
+        <app-application-workflow-item [project]="project"  [workflowItem]="item" [orientation]="orientation" [application]="application" [applicationFilter]="applicationFilter" [branch]="branch"></app-application-workflow-item>
```
Comment (sguiheux) [inline ui/src/app/views/application/show/workflow/tree/application.tree.workflow.html]: no need branch

File: ui/src/app/views/application/show/workflow/tree/application.tree.workflow.html [hunk 4]
```diff
@@ -1,6 +1,6 @@
...
+        <app-application-tree-workflow *ngIf="item.subPipelines" [project]="project"  [workflowItems]="item.subPipelines" [orientation]="orientation" [application]="application" [applicationFilter]="applicationFilter" [branch]="branch"></app-application-tree-workflow>
```
Comment (sguiheux) [inline ui/src/app/views/application/show/workflow/tree/application.tree.workflow.html]: no need branch

File: ui/src/app/views/application/show/workflow/tree/application.tree.workflow.component.ts [hunk 5]
```diff
@@ -13,6 +13,7 @@ export class ApplicationTreeWorkflowComponent {
     @Input() application: Application;
     @Input() workflowItems: Array<WorkflowItem>;
     @Input() orientation: string;
+    @Input() branch: string;
```
Comment (sguiheux) [inline ui/src/app/views/application/show/workflow/tree/application.tree.workflow.component.ts]: no need, value is on applicationFilter

File: ui/src/app/views/application/show/workflow/tree/item/application.workflow.item.component.ts (hunk 1 above)
Comment (bnjjj) [inline ui/src/app/views/application/show/workflow/tree/item/application.workflow.item.component.ts]: When you have a pipeline building and change the branch filter without this condition it will send a false notification

File: ui/src/app/views/application/show/workflow/tree/item/application.workflow.item.component.ts [hunk 6]
```diff
@@ -34,9 +33,11 @@ export class ApplicationWorkflowItemComponent implements DoCheck {
     @Input() project: Project;
     @Input() workflowItem: WorkflowItem;
     @Input() orientation: string;
+    @Input() branch: string;
```
Comment (sguiheux) [inline ui/src/app/views/application/show/workflow/tree/item/application.workflow.item.component.ts]: no need

File: ui/src/app/views/application/show/workflow/application.workflow.html (hunk 2 above)
Comment (bnjjj) [inline ui/src/app/views/application/show/workflow/application.workflow.html]: Oups ! You're right ! My bad
